File: 4.Arrays Level-0/4.3 Reverse String/reverse-string.py
#Reverse String - https://leetcode.com/problems/reverse-string/

# Write a function that reverses a string. The input string is given as an array of characters s.
# You must do this by modifying the input array in-place with O(1) extra memory.

def reverse_string(s):
    for i in range(0,int(len(s)/2)):
        temp = s[i]
        s[i] = s[len(s)-1-i]
        s[len(s)-1-i] = temp
    return s

print(reverse_string(["h", "e", "l", "l", "o"]))
print(reverse_string(["H", "a", "n", "n", "a", "h"]))

File: 4.Arrays Level-0/4.3 Reverse String/reverse-string.js
// Reverse String - https://leetcode.com/problems/reverse-string/

// Write a function that reverses a string. The input string is given as an array of characters s.
// You must do this by modifying the input array in-place with O(1) extra memory.

function reverseString(s) {
  let len = Math.floor(s.length / 2);
  for (let i = 0; i < len; i++) {
    let temp = s[i];
    s[i] = s[s.length - 1 - i];
    s[s.length - 1 - i] = temp;
  }
  return s;
}

console.log(reverseString(["h", "e", "l", "l", "o"]));
console.log(reverseString(["H", "a", "n", "n", "a", "h"]));
